jax/experimental/export/export.py

```diff
@@ -595,33 +634,58 @@ def is_token(attrs):
     # The order of args: platform_index_arg, dim args, token args, array args.
     nr_platform_index_args = 1 if has_platform_index_argument else 0
     nr_dim_args = len(dim_vars)
-    nr_token_args = sum(1 for attrs in arg_attrs if is_token(attrs))
-    nr_array_args = len(orig_input_types) - nr_platform_index_args - nr_dim_args - nr_token_args
+    token_arg_idxs = [i for i, (typ, attrs) in enumerate(zip(orig_input_types,
```

Comment:
I wanted the export function to produce the StableHLO as "natively" as possible so that when called from JAX it composes perfectly. JAX uses separate tokens for different effects and this lowering allows us to connect the tokens properly when we call the exported module.

When we call the module from TF, through XlaCallModule I will implent the scheme you proposed.